codecov.yml

```diff
@@ -3,6 +3,11 @@ comment:                  # this is a top-level key
   behavior: default
   require_changes: true  # if true: only post the comment if coverage changes
 
+#list of paths (folders or file names) to exclude files from being collected by Codecov
 ignore:
 - "test"
-- "apis"
\ No newline at end of file
+- "controllers/infrastructure/byohost_controller.go"
+- "controllers/infrastructure/byomachinetemplate_controller.go"
+- "apis/infrastructure/v1beta1/*_types.go"
+- "apis/infrastructure/v1beta1/zz_generated.deepcopy.go"
+- "agent/installer/cli.go"
```

Comment:
We have actual code in the `cli.go` right? This probably shouldn't be ignored?

Reply:
This [Issue](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/issues/344) for improving agent/installer code test coverage was closed and It was found that `cli.go` is hard to test/mock[[here](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/issues/341#issuecomment-1027691657)] and hence we are excluding it from the code coverage.

@anusha94 @NilanjanDaw can add more.

Reply:
lets open an issue to simplify code-design, wherever applicable for testability and address this overall not just cli.go
this PR can then proceed
@jamiemonserrate wdyt?

Reply:
Disclaimer - I am not saying no for an issue to simplify code-design.

We have 3 issues to cover existing code refactor / improvements / coverage - inputs on those issues / PRs would be helpful. See linked PRs in below issues
- https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/issues/221
- https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/issues/254
- https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/issues/340

With this, we hope to clear any tech debt we had accumulated at the time of v0.1.0 release. And this should reasonably improve the overall health and coverage. 
Coming to `cli.go` - this is used only to test the installer locally and is only an optional dev workflow. The functions that are called by `cli.go` are already covered by other unit / integration tests in the package. So this leaves us with writing tests for the flags supported by `cli.go`. We could most definitely do that, but didn't find much value add. And including this file in test coverage report will give a false indication of the overall health of the package.

So I had asked Nilanjan to mark this for exclusion. But if y'all think this needs to be tested for flags or any other use-case of `cli.go` I'm missing, happy to write tests. Then it makes sense to not exclude this file in codecov.

Reply:
> Coming to cli.go - this is used only to test the installer locally and is only an optional dev workflow.
I guess then we should name it or put it in test package and call it out to show only a dev/test flow

+1 if the underlying code is already covered in other flows but the test coverage doesn't show that https://app.codecov.io/gh/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/agent/installer/cli.go

Reply:
> I guess then we should name it or put it in test package and call it out to show only a dev/test flow

Created https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/issues/374 to track this